untyped-lambda-calculus/article.md

```diff
@@ -146,63 +146,73 @@ This brings us to the discussion of evaluation strategies.
 
 An evaluation strategy is a rule that defines which redexes are reduced and in what order.
 
-The simplest approach would be to not care about the order, just reduce any random redex until there's nothing left.
+The simplest approach is to reduce redexes in any random order until there's nothing left.
 This approach is called _full beta-reduction._
-It does have some issues, however.
+But it has some issues.
 For instance, you might get different results depending on the reduction order.
 
-So it stands to reason to specify the reduction order.
-There are multiple ways we can do that.
+So it's better to reduce in a specific order.
+There are multiple ways to approach this. 
 
-First, we need to decide whether we're going left-to-right or right-to-left.
-This decision happens to be easy: going right-to-left is strictly less powerful, i.e. the process doesn't terminate on some terms whereas going left-to-right does.
+Our decision points are: 
+* to start with the left or right side of the expression;
+* to start with the outermost or the innermost redex;
+* to reduce redexes inside abstraction bodies or not.
 
-Second, we'll decide whether we're first reducing the outermost or the innermost redex.
-To see what is meant by outermost and innermost, consider the following expression:
+#### Direction of reducing
+
+We can reduce an expression left-to-right or right-to-left.
+This decision is easy: going right-to-left is strictly less powerful because the process doesn't terminate on some terms whereas going left-to-right does.
```

Comment:
Okay, my fault, but this doesn't sound right. What this meant to say is probably

> going right-to-left is strictly less powerful because the process doesn't terminate on all terms going left-to-right does, but not vice versa.

Reply:
Updated